python-package/xgboost/core.py

```diff
@@ -1603,7 +1609,7 @@ def attr(self, key: str) -> Optional[str]:
             return py_str(ret.value)
         return None
 
-    def attributes(self) -> Dict[str, str]:
+    def attributes(self) -> Dict[str, Optional[str]]:
```

Comment:
I think dict[str, str] should be correct. Have you found a case where only the key is available?

Reply:
Yes, the case is right above this function at line 1591: 
`def attr(self, key: str) -> Optional[str]:`
`attr(key)` is called in `attributes()` at line 1626:
`return {n: self.attr(n) for n in attr_names}`